File: scratch/check_divs.js
const fs = require('fs');
const html = fs.readFileSync('c:/Users/72671/Desktop/שפות תכנות/סיכום/3.html', 'utf8');
const lines = html.split('\n');
let openDivs = 0;
lines.forEach((line, i) => {
    const openCount = (line.match(/<div/g) || []).length;
    const closeCount = (line.match(/<\/div>/g) || []).length;
    if (line.includes('class="tab-content')) {
        console.log(`Tab started at line ${i+1} with depth ${openDivs}`);
    }
    openDivs += openCount - closeCount;
    if (line.includes('</main>')) {
        console.log(`</main> found at line ${i+1} with depth ${openDivs}`);
    }
});
console.log(`Final depth: ${openDivs}`);

File: scratch/check_divs.py
import sys
with open('c:/Users/72671/Desktop/שפות תכנות/סיכום/3.html', 'r', encoding='utf-8') as f:
    html = f.read()

lines = html.split('\n')
open_divs = 0
for i, line in enumerate(lines):
    open_count = line.count('<div')
    close_count = line.count('</div')
    if 'class="tab-content' in line:
        print(f"Tab started at line {i+1} with depth {open_divs}")
    open_divs += open_count - close_count

print(f"Final depth: {open_divs}")
